crates/starknet/src/error.rs

```diff
@@ -0,0 +1,26 @@
+use starknet_types;
+
+use starknet_in_rust;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    StarknetApiError(#[from] starknet_api::StarknetApiError),
+    #[error(transparent)]
+    StateError(#[from] starknet_in_rust::core::errors::state_errors::StateError),
+    #[error(transparent)]
+    TransactionError(#[from] starknet_in_rust::transaction::error::TransactionError),
+    #[error("Types error")]
+    TypesError(#[from] starknet_types::error::Error),
+    #[error("Specifying block by hash is currently not enabled")]
+    BlockIdHashUnimplementedError,
+    #[error("Specifying block by number is currently not enabled")]
+    BlockIdNumberUnimplementedError,
+    #[error("I/O error")]
+    IoError(#[from] std::io::Error),
+    #[error("Error when reading file {path}")]
+    ReadFileError { source: std::io::Error, path: String },
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
```

Comment:
How about keeping the name `DevnetResult` to avoid confusion with `std::result::Result` that might be expected when you see `Result`?